momento-provider/src/test/java/com/google/code/ssm/providers/momento/TestPOC.java

```diff
@@ -0,0 +1,122 @@
+package com.google.code.ssm.providers.momento;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.code.ssm.Cache;
+import com.google.code.ssm.CacheFactory;
+import com.google.code.ssm.api.ParameterValueKeyProvider;
+import com.google.code.ssm.api.ReadThroughSingleCache;
+import com.google.code.ssm.api.format.SerializationType;
+import com.google.code.ssm.config.AbstractSSMConfiguration;
+import com.google.code.ssm.config.DefaultAddressProvider;
+import lombok.Data;
+import lombok.SneakyThrows;
+import lombok.ToString;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.io.Serializable;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(loader= AnnotationConfigContextLoader.class)
+public class TestPOC {
+    @Configuration
+    public static class ContextConfiguration extends AbstractSSMConfiguration {
+        @Bean
+        @Override
+        public CacheFactory defaultMemcachedClient() {
+            final MomentoConfiguration conf = new MomentoConfiguration();
+            conf.setConsistentHashing(true);
+            conf.setDefaultTtl(300);
+            // IMPORTANT: This presumes you have already created a cache in Momento named 'example-cache'
+            conf.setDefaultCacheName("example-cache");
+            conf.setMomentoAuthToken("<YOUR AUTH TOKEN GOES HERE>");
+            final CacheFactory cf = new CacheFactory();
+            cf.setCacheClientFactory(new MomentoClientFactoryImpl());
+            // Doesn't actually do anything
+            cf.setAddressProvider(new DefaultAddressProvider("127.0.0.1:11211"));
+            cf.setConfiguration(conf);
+            return cf;
+        }
+    }
+    @Autowired
+    private CacheFactory cacheFactory;
+
+    @Data
+    @JsonSerialize
+    public static class TestObject implements Serializable {
+        int id;
+        String name;
+    }
+
+
+    @ReadThroughSingleCache(namespace = "TestString", expiration = 3600)
```

Comment:
what does the namespace do here?

Also it looks like this presumes a single `CacheFactory` bean with a single cache (which your annotation does imply, so that's good), but do we know if that is sufficient for them, or what it would take to support specifying the cache name in the annotation here?

Reply:
> what does the namespace do here

I believe you can have different caches set up based on a namespace

> what it would take to support specifying the cache name in the annotation here?
I'm not sure what it would take to support tbh, but I think we need a clearer use case from the customer before crossing that bridge